### PR TITLE
Add search, sort and pagination to probe listing page

### PR DESCRIPTION
**Pull request by lzeee**
Due to the unreasonable process of my previous PR, I opened this PR and kept the submission history of the original contributors. Thanks for your contribution, the valuable historical code review comments and the patient guidance! These led me to gradually learn how to contribute to the community. @dheerajodha @alecharp 

### History background
It's an old issue https://github.com/jenkins-infra/plugin-health-scoring/issues/65. What I have done is just trying to add the necessary packages to the system and make it work. **90% of the codes** comes from 2 historical PR** made by https://github.com/jenkins-infra/plugin-health-scoring/pull/109 @dheerajodha and https://github.com/jenkins-infra/plugin-health-scoring/pull/194 @impact-maker. 

### Description
This PR is an enhancement to the display page of the probe, allowing you to page, sort and search the probes. In order to accomplish this goal, I mainly did these things based on the historical code:
1. Add 3 npm packages(datatables.net、jquery、expose-loader) and make them work. Specify the '/probe' page when processing the table.
2. Add 1 *.less files to control the style of 4 added component, which is supplied by datatables.net.
3. Modify several probe classes to adapt to the newly added abstract function.

I think my code needs serious review, any suggestions and comments would be greatly appreciated !

the screen shot:
show

### Submitter checklist
- [ ] The description of this pull request is detailed and explain why this pull request is needed
- [ ] The changeset is on a specific branch
  - `feature/` for new feature, or improvements
  - `fix/` for bug fixes
  - `docs/` for any documentation changes
- [ ] If required, the documentation has been updated
- [ ] There is automated tests to cover the code change / addition
    - If there is no test, include an explanation why in the description
- [ ] Run `mvn verify` locally and all tests are passing successfully
  - It is OK to create a pull request which has failing tests if it is created as a draft, is to fix a bug and the first commit is the test to prove the existence of the bug.
- [ ] There is no new warnings (checkstyle nor spotbugs) on the code

### CODE REVIEW

File: src/main/js/index.js [hunk 1]
```diff
@@ -2,3 +2,13 @@ import { defineCustomElements } from "ionicons/dist/loader";
...
 });
+
+import $ from 'jquery';
+import dt from 'datatables.net';
+$(document).ready( function () {
+  if (window.location.pathname === "/probes") {
+    $('#probes-table').DataTable(
+      {"order":[[3, 'asc']]}
+    );
+  }
```
Comment (alecharp) [inline src/main/js/index.js:13]: This simple complex. Can we not have a `<script>` in the correct template and not have this in the global javascript?

File: package.json [hunk 2]
```diff
@@ -12,6 +12,9 @@
     "copy-webpack-plugin": "^11.0.0",
     "css-loader": "^6.7.3",
     "css-minimizer-webpack-plugin": "^4.2.2",
+    "datatables.net": "^1.13.2",
```
Comment (alecharp) [inline package.json:15]: Shouldn't this be a `dependency` and not a `devDependency`?

File: package.json [hunk 3]
```diff
@@ -12,6 +12,9 @@
...
+    "expose-loader": "^4.0.0",
+    "jquery": "^3.6.3",
```
Comment (alecharp) [inline package.json:17]: same question.

**Review by alecharp — COMMENTED** [review]
I haven't tested locally yet but please see my comment here.

File: src/main/js/index.js (hunk 1 above)
Comment (lzeee) [reply src/main/js/index.js:13]: Sorry for the late reply, I have been writing the draft proposal with all my strength these two days.

I see what you mean, I'm going to try it further, I also find it weird to put it here

File: package.json (hunk 2 above)
Comment (lzeee) [reply package.json]: I moved the jquery and the datatables to the  dependency .

File: package.json (hunk 3 above)
Comment (lzeee) [reply package.json]: I moved the jquery and the datatables to the  dependency .

**Comment by lzeee** [comment]
@alecharp 
Hi, I found a way to avoid adding a <script> to the template and removing global function, using the webpack plugin named html-webpack-plugin. I'm not sure whether it's a reasonable way, you can review it when you have time : )

I made these changes to make it work. Most of them are in the file webpack.config.js.

1.add an entry to generate the listing.js file
2.add html-webpack-plugin
3.use html-webpack-plugin to inject the listing.js file to the template 'score/listing.html'

There it works well, but the splitChunks will import two vendors.js to listing.html. Because it's imported to main.html as default. So I did this:
4.remove listing.js when doing the optimization.splitChunks

Then it's done. I also tried to modify some styles, the following is the screenshot:

show

**Pull request closed without merging**